### ISSUE TITLE
Playground block: Save manual website customizations in the post 

### ISSUE THREAD

**Issue by adamziel** [WordPress/playground-tools#166]
@westnz said:

> I wanted to confirm, if I add the Playground block, I can add dummy content etc. and then press Save.  When a learner interacts with the block on the front end they will be able to use the dummy content I added to complete the task or recreate an example? 

This is not how the block works today. You can specify a custom Blueprint to set up the site, which requires writing custom JSON.

The customizations done through the UI are not saved, but it's such a great idea tTe block should absolutely support that. I don’t see any technical blockers for building it. Technically, saving a post could export Playground as zip file, upload that zip as a media file, and then use that zip to import content the next time Playground is loaded.

Here's a few examples of possible usage in the tutorials from @westnz:

- Setting up a homepage with a header, and asking someone to re-create a footer example.
- Adding media and asking someone to re-create a pattern / group of nested blocks or even 404 template.
- Creating a spam message and asking someone to tag it as spam and updating the Disallowed Comment Keys section.
- Setting up a Contact page and then asking a learner to install a forms plugin + add a contact form
- Setting up homepage and then asking someone to use global styles to update certain aspects of the design.

@bgrgicak

**Comment by bgrgicak** [comment]
Yes, if we publish the Playground plugin, we could automatically install it and use it in the background to create snapshots that would be loaded in Playground.

When an editor makes changes on the post-edit screens and clicks update, we could trigger the snapshot process.
On the post page users would load that data. We could even enable browser storage by default to save the state on post pages for the user.

**Comment by bgrgicak** [comment]
@adamziel I'm happy to work on this. But I wouldn't be able to commit to it this week.

**Comment by adamziel** [comment]
@bgrgicak let's hold on for now and:

* Gather more feedback in https://github.com/WordPress/Learn/pull/2264 – perhaps that wouldn't be blocking for the learn team, while the work on logging and fixing Playground errors unblocks people for sure.
* Give PHP Blueprints a bit more time to become stable and support WordPress export/import – this could solve for the problem of getting the data out and into Playground. There is an existing export feature, but it has quite a few rough and weird behaviors that will be much easier to iron out in PHP.
